elliptic-curve/src/lib.rs

```diff
@@ -21,21 +21,30 @@
 #[cfg(feature = "std")]
 extern crate std;
 
-#[cfg(feature = "rand_core")]
-pub use rand_core;
-
 pub mod error;
 pub mod secret_key;
 
-pub use generic_array::{self, typenum::consts};
-pub use subtle;
-
 // TODO(tarcieri): other curve forms
 #[cfg(feature = "weierstrass")]
 #[cfg_attr(docsrs, doc(cfg(feature = "weierstrass")))]
 pub mod weierstrass;
 
 pub use self::{error::Error, secret_key::SecretKey};
+pub use generic_array::{self, typenum::consts};
+pub use subtle;
+
+#[cfg(feature = "rand_core")]
+pub use rand_core;
+
+#[cfg(feature = "rand_core")]
+use rand_core::{CryptoRng, RngCore};
 
 /// Byte array containing a serialized scalar value (i.e. an integer)
 pub type ScalarBytes<Size> = generic_array::GenericArray<u8, Size>;
+
+/// Trait for randomly generating a value
+#[cfg(feature = "rand_core")]
+pub trait Generate {
+    /// Generate a random element of this type using the provided [`CryptoRng`]
+    fn generate(rng: &mut (impl CryptoRng + RngCore)) -> Self;
```

Comment:
`RngCore` is implemented for mutable RNG references, so we can use `rng: impl CryptoRng + RngCore` here.

Reply:
Ooh, nice